src/constants.zig

```diff
@@ -53,9 +53,9 @@ comptime {
 
 /// The checkpoint interval is chosen to be the highest possible value that satisfies the
 /// constraints described below.
-pub const vsr_checkpoint_interval = journal_slot_count -
-    lsm_batch_multiple -
-    lsm_batch_multiple * stdx.div_ceil(pipeline_prepare_queue_max * 2, lsm_batch_multiple);
+pub const vsr_checkpoint_interval = journal_slot_count - lsm_compaction_interval *
+    // Trigger plus two pipelines, rounded up to `lsm_compaction_interval`.
+    (1 + stdx.div_ceil(pipeline_prepare_queue_max * 2, lsm_compaction_interval));
```

Comment:
I am _almost_ motivated to introduce

```
stdx.round_up_to(value, multiple)
```

or

```
stdx.next_multiple(value, multiple)
```

Reply:
`next_multiple` is 🔥